### PR TITLE
KB migration: Restore an environment

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) adds a new task topic to the cloud guide for restoring an environment. It migrates [this KB article](https://support.magento.com/hc/en-us/articles/360000852534-Reset-environment-on-Cloud) to devdocs.

See MAGEDOC-4000.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.2/cloud/env/restore-environment.html
-  https://devdocs.magento.com/guides/v2.3/cloud/env/restore-environment.html
- https://devdocs.magento.com/guides/v2.2/cloud/project/project-webint-snap.html
- https://devdocs.magento.com/guides/v2.3/cloud/project/project-webint-snap.html

whatsnew
Added a new topic about [restoring a Magento Commerce Cloud environment](https://devdocs.magento.com/guides/v2.3/cloud/env/restore-environment.html).

### CODE REVIEW

File: guides/v2.2/cloud/env/restore-environment.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: Restore an environment
+group: cloud-guide
+---
+
+If you encounter issues in your environment, but you do not have a [valid snapshot]({{ page.baseurl }}/cloud/project/project-webint-snap.html), you can try restoring your environment in on of the following ways:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
If you encounter issues in your environment, but you do not have a [valid snapshot]({{ page.baseurl }}/cloud/project/project-webint-snap.html), you can try restoring your environment in one of the following ways:
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+-  SSH
+-  Git
+
+## SSH
+
+If you have not created a snapshot but you can access the environment using SSH, follow these steps to restore your environment:
+
+-  Disable configuration management
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]: Should they be numbered? Is the order important?

File: guides/v2.2/cloud/env/restore-environment.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+-  Returns your Magento installation to its original state by restoring the database, removing the deployment configuration, and clearing all directories under `var/`.
+-  Resets your git branch to a stable state in the past.
+
+### Disable configuration management
+
+You must disable [configuration management]({{ page.baseurl }}/cloud/live/sens-data-over.html) so that it does not automatically apply the previous configuration settings during deployment.
+
+To disable configuration management, make sure that your `app/etc/` directory does not contain the `config.php` file.
+
+To remove the configuration file:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]: procedure title class?
{:.procedure}

File: guides/v2.2/cloud/env/restore-environment.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+   -  For Magento 2.1:
+
+      ```bash
+      rm app/etc/config.local.php
+      ```
+
+### Uninstall Magento
+
+Uninstalling the Magento software drops and restores the database, removes the deployment configuration, and clears directories under `var/`.
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]: Wouldn't those be `var/` subdirectories?

File: guides/v2.2/cloud/env/restore-environment.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+To uninstall the Magento software:
+
+1. [SSH to your environment]({{ page.baseurl }}/cloud/env/environments-ssh.html#ssh).
+1. Run the uninstall command:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Uninstall the Magento application.
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+   git log --reverse
+   ```
+
+1. Choose a commit hash that represents that last known stable state of your code.
+
+   To reset your branch to its original initialized state, find the very first commit that created your branch.
+
+   ![Select a commit hash in the git console]({{ site.baseurl }}/common/images/select-commit-hash.png)
+
+1. Reset your branch:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Use the hard reset option to reset your branch.
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+```bash
+git commit --allow-empty -m "<message>" && git push <origin> <branch>
+```
+
+## Uninstall failed
+
+If executing the `setup:uninstall` command fails and cannot be completed, you may need to manually reset the database:
+
+1. [SSH to your environment]({{ page.baseurl }}/cloud/env/environments-ssh.html#ssh).
+1. Connect to the database:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Connect to the database.
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+   ```bash
+   mysql -h database.internal
+   ```
+
+1. Drop the `main` database:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Drop the `main` database.
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+   ```shell
+   drop database main;
+   ```
+
+1. Create an empty `main` database:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Create an empty `main` database.
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+   ```shell
+   create database main;
+   ```
+
+1. Delete the following configuration files:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Delete the following configuration files.
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+
+      ```bash
+      rm app/etc/config.local.php
+      ```
+
+### Uninstall Magento
+
+Uninstalling the Magento software drops and restores the database, removes the deployment configuration, and clears directories under `var/`.
+
+To uninstall the Magento software:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]: procedure title class

File: guides/v2.2/cloud/env/restore-environment.md [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+
+   ```terminal
+   [SUCCESS]: Magento uninstallation complete.
+   ```
+
+### Reset the git branch
+
+Resetting your branch reverts the code to a stable state in the past.
+
+To reset your branch:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]: procedure title class

File: guides/v2.2/cloud/env/restore-environment.md [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+
+   ![Select a commit hash in the git console]({{ site.baseurl }}/common/images/select-commit-hash.png)
+
+1. Reset your branch:
+
+   ```bash
+   git reset --h <commit_hash>
+   ```
+
+1. Push your changes to trigger a redeploy, which reinstalls Magento:
```
Comment (hguthrie) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
1. Push your changes to trigger a redeploy, which reinstalls Magento.
```

File: guides/v2.2/cloud/env/restore-environment.md (hunk 2 above)
Comment (jeff-matthews) [reply guides/v2.2/cloud/env/restore-environment.md]: Yes, I think so. The original KB article numbers them, so I'll update it.

File: guides/v2.2/cloud/env/restore-environment.md (hunk 3 above)
Comment (jeff-matthews) [reply guides/v2.2/cloud/env/restore-environment.md]: Why not? ;)

File: guides/v2.2/cloud/env/restore-environment.md (hunk 4 above)
Comment (jeff-matthews) [reply guides/v2.2/cloud/env/restore-environment.md]: Indeed

File: guides/v2.2/cloud/env/restore-environment.md (hunk 11 above)
Comment (jeff-matthews) [reply guides/v2.2/cloud/env/restore-environment.md]: will do

File: guides/v2.2/cloud/env/restore-environment.md (hunk 12 above)
Comment (jeff-matthews) [reply guides/v2.2/cloud/env/restore-environment.md]: yes

File: guides/v2.2/cloud/env/restore-environment.md [hunk 14]
```diff
@@ -0,0 +1,146 @@
+---
+title: Restore an environment
+group: cloud-guide
+---
+
+If you encounter issues in your environment, but you do not have a [valid snapshot]({{ page.baseurl }}/cloud/project/project-webint-snap.html), you can try restoring your environment in one of the following ways:
```
Comment (meker12) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
If you encounter issues in your environment and do not have a [valid snapshot]({{ page.baseurl }}/cloud/project/project-webint-snap.html), you can try restoring your environment in one of the following ways:
```

File: guides/v2.2/cloud/env/restore-environment.md [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+
+-  SSH
+-  Git
+
+## SSH
+
+If you have not created a snapshot but you can access the environment using SSH, follow these steps to restore your environment:
```
Comment (meker12) [inline guides/v2.2/cloud/env/restore-environment.md]:
```suggestion
If you have not created a snapshot, but you can access the environment using SSH, follow these steps to restore your environment:
```

**Comment by jeff-matthews** [comment]
Thanks @meker12.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**